### ISSUE TITLE
[WCAG 2.2 Draft Feedback] Success Criterion 2.4.13 Focus Not Obscured (Level AAA)

### ISSUE THREAD

**Issue by dshoukry** [w3c/wcag#2711]
“Success Criterion 2.4.13 Focus Not Obscured (Enhanced) (Level AAA): When a user interface component receives keyboard focus, no part of the focus indicator is hidden by author-created content.”

Recommendation:
Most of our comments/proposals are related to requests posed in the AA counterpart, it may be worth trying to clarify those first and then update this document accordingly. 

We’d also like to understand more about the intended differences between the AA requiring component visibility and the AAA requiring that the focus indicator remains completely visible. In the AAA SC, "The purpose of this Success Criterion is to ensure that a component with keyboard focus is visible. This criterion is closely related to Focus Not Obscured (Minimum) but requires that the whole of the component is visible." [emphasis ours]. That doesn't match the SC itself that requires full focus indicator visibility.

Please find detailed specifics covered in our [Sep22 2.4.13 Focus Not Obscured (AAA) Google Doc](https://docs.google.com/document/d/1JJMn_zEYxTU3xhq_4o9PEjI1rlFNkwuGjcbT5WUgFtQ/edit)

**Comment by mbgower** [comment]
https://github.com/w3c/wcag/issues/2736 also covers one of the key concerns expressed in this issue

**Comment by bruce-usab** [comment]
@dshoukry, as you note, AA 2.4.12 is about the component while AAA 2.4.13 is about the focus indicator.

The archetype situation is a visually-oriented person who does not use a mouse, and is using the tab key to navigate through a page.

A largish visual pop-up that happens to cover most of the original component, and even the focus indicator disappears, probably is not so disorienting as to cause the person to loose their place, since AA SC 2.1.12  requires that portion of the original component remain visible.

AAA SC 2.4.13 goes a step further, by requiring that the entire focus indicator remain visible.  This provides that even if someone is not watching the screen while tabbing (maybe they are looking down at their keyboard while tabbing), it is easy to notice where is the keyboard focus.  Having the entire focus indicator visible _probably_ means that the whole of the component is visible, but the focus indicator itself might cover significant portions of the component.

I agree that the linked Understanding documents need to be updated some.

**Comment by alastc** [comment]
Response for group consideration, focusing on the comments unique to the AAA SC:

<hr>

> Can the SC match the Understanding documentation instead of having such sweeping guidance that goes well beyond the intended scope?

The understanding document and SC do need aligning, that has been started in #2807. The following questions are more specific about the scope.

> If there is even a single pixel of the focus indicator not visible, it would pass 2.4.12 and fail 2.4.13. Is this correct?
> Is there a rationale that concludes how a pixel being obscured would be a barrier to users?

The core rational for both is the same, that the focused element and it's indicator should be visible to the user.

What varies is the difficulty in achieving that, and difficulty of testing. 

It is quite easy for a focus indicator to be somewhat obscured by adjacent content, e.g. the indicator goes underneath the adjacent item. Not enough to fail focus-appearance, but partially obscured. That isn't hugely problematic, at least compared to when the entire element & focus indicator are hidden behind other content.

So the AA version is intended to catch the worst cases, whilst allowing for minor issues. It is easier to ensure the element is visible than the focus indicator.

The AAA version is the ideal, where the focus indicator is never affected by other authored content.

We did consider allowing for some percentage of the indicator to be hidden, but it massively increases the complexity of testing. Given that it is at the AAA level, it is more straightforward to require zero overlap.


> If a component has an increased visibility focus indicator, such as large text or a much bolder focus ring, wouldn't that make that component much more likely to fail this criterion as these indicators take up more space and are more likely to be partially obscured?

Possibly, it often comes down to the z-index aspects in HTML. However, this is the AAA "ideal" version of the requirement. 

> It'd be very helpful to have at least one more example here

Another example has been added in #2807.

> What are the expectations around the popular use of non-modal dialogs and the growing popularity of overlaying content such as chatbots and sticky elements? There are many libraries that open non-modal dialogs based on the available size in the viewport, however these are not likely taking focus indicators into consideration.
> Since no part of the focus outline can be obscured as part of this criterion, there will be many failures and enforcement may be very difficult.

Indeed, at the AAA level those features can be difficult depending on how they are implemented. This was considered, and the AA level version is more forgiving because:

- It focuses on the 'when receiving focus', rather than at any time with focus.
- It allows for some part to be hidden.

In the techniques:
> It'd be very helpful to have at least two examples here

Is there another scenario that would be useful to cover? The main target was sticky elements.

Adding techniques & failures is an ongoing task.

**Comment by mbgower** [comment]
@alastc 

> Another example has been added in https://github.com/w3c/wcag/pull/2807.

No image has not been added to the PR, as far as I can see.

> the AA level version is more forgiving because It focuses on the 'when receiving focus', rather than at any time with focus.

That's wasn't true in the previously published version. Cuurrently in latest editor's both AA and AAA are shown as being "when receiving focus" instead of "...when a UIC has focus..." When was that changed? I'm very concerned this got altered because it means someone could have sufficient contrast for a millisecond and then make the contrast utterly disappear. 

I assume this was brought in because there was a concern expressed that enhanced indicators should be allowed to fade after awhile. If they are truly enhanced and fall back to this minimum, that's fine. But if we don't specify a minimum, I don't see how the "receives" language can be considered adequate.

**Comment by alastc** [comment]
Hi @mbgower,

> Currently in latest editor's both AA and AAA are shown as being "when receiving focus" instead of "...when a UIC has focus..." When was that changed? I'm very concerned this got altered because it means someone could have sufficient contrast for a millisecond and then make the contrast utterly disappear.

I can't remember when it changed, but remember this isn't focus-appearance, this is about the focused element not being obscured by other content. If that doesn't happen when it received focus, it should be a problem (at least for the scenarios we were aiming to cover).

**Comment by mbgower** [comment]
> remember this isn't focus-appearance, this is about the focused element not being obscured by other content. If that doesn't happen when it received focus, it should be a problem (at least for the scenarios we were aiming to cover)

Right, thanks for that clarity. Okay, that's much less of a concern then, and may in fact be a sensible constriction, since it is when it receives focus that we care about this the most.

**Comment by alastc** [comment]
The group approved the response above, so closing this issue:
https://www.w3.org/2023/02/07-ag-minutes#item11

**Comment by dshoukry** [comment]
> #2807
If a component has an increased visibility focus indicator, such as large text or a much bolder focus ring, wouldn't that make that component much more likely to fail this criterion as these indicators take up more space and are more likely to be partially obscured?

>>> Possibly, it often comes down to the z-index aspects in HTML. However, this is the AAA "ideal" version of the requirement.

The person who shared this feedback wanted to note that: The z-index approach can be useful for non-static positioned elements, but for static elements it wouldn't help as this property has no effect. They are concerned that this AAA criterion can discourage the use of larger, more visible focus indicators (such as increased size, larger text, highlighted background, etc.), as it can increase the likelihood that some of the indicator could be covered by elements with hidden overflow or negative margin (as examples).

**Comment by mbgower** [comment]
@dshoukry please note that the language of the AAA criterion was changed so that it is now about the component being obscured, not the indicator.

> When a [user interface component](https://www.w3.org/TR/WCAG22/#dfn-user-interface-components) receives keyboard focus, no part of the component is hidden by author-created content.
